Update file-4.x connector doc (#419)
wso2/docs-mi: en/docs/reference/connectors/file-connector/file-connector-example.md

* `/create `: The user sends the request payload which includes the location where the file needs to be saved and the content needs to be added to the file. This request is sent to the integration runtime by invoking the FileConnector API. It saves the file in the specified location with the relevant content.

    <p><img src="{{base_path}}/assets/img/integrate/connectors/filecon-3.x/fileconnector-03.png" title="Adding a Rest API" width="800" alt="Adding a Rest API" /></p>

* `/read `: The user sends the request payload, which includes the location of the file that needs to be read. This request is sent to the integration runtime where the FileConnector API resides. Once the API is invoked, it first checks if the file exists in the specified location. If it exists, the content is read and response is sent to the user. If the file does not exist, it sends an error response to the user.

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon-3.x/fileconnector-02.png" title="Adding a Rest API" width="800" alt="Adding a Rest API"/>

If you do not want to configure this yourself, you can simply [get the project](#get-the-project) and run it.

> ## Configure the connector in WSO2 Integration Studio

Reviewer: ```suggestion
## Setting Up the Integration Project
```

> Follow these steps to set up the Integration Project and the Connector Exporter Project.

Reviewer: ```suggestion
Follow these steps to set up the Integration Project using the WSO2 Micro Integrator Visual Studio Code extension.
```

## What you'll build

> This example explains how to use File Connector to create a file in the local file system and read the particular file. The user sends a payload specifying which content is to be written to the file. Based on that content, a file is created in the specified location. Then the content of the file can be read as an HTTP response by invoking the other API resource upon the existence of the file.

Reviewer 2: Let's keep 'Similarly, the same example can be configured to communicate with a remote file system (i.e FTP server) easily. The example also uses some other WSO2 mediators to manipulate messages. '

* `/create `: The user sends the request payload which includes the location where the file needs to be saved and the content needs to be added to the file. This request is sent to the integration runtime by invoking the FileConnector API. It saves the file in the specified location with the relevant content.

    <p><img src="{{base_path}}/assets/img/integrate/connectors/filecon-3.x/fileconnector-03.png" title="Adding a Rest API" width="800" alt="Adding a Rest API" /></p>

* `/read `: The user sends the request payload, which includes the location of the file that needs to be read. This request is sent to the integration runtime where the FileConnector API resides. Once the API is invoked, it first checks if the file exists in the specified location. If it exists, the content is read and response is sent to the user. If the file does not exist, it sends an error response to the user.

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon-3.x/fileconnector-02.png" title="Adding a Rest API" width="800" alt="Adding a Rest API"/>

If you do not want to configure this yourself, you can simply [get the project](#get-the-project) and run it.

> ## Setting Up the Integration Project

Reviewer 2: ```suggestion
## Set up the integration project
```

Follow these steps to set up the Integration Project using the WSO2 Micro Integrator Visual Studio Code extension.

### Creating the New Project

1. Go to **WSO2 Micro Integrator** in the VS Code.

2. Click on **Create New Project** to create the new integration project.

3. provide the **Project Name** and select the **Project Directory**

    <img src="{{base_path}}/assets/img/integrate/connectors/new-vscode-project.png" title="Adding a Rest API" width="500" alt="Adding a Rest API"/>

4. You have now created the new project with the following structure.

     <img src="{{base_path}}/assets/img/integrate/connectors/prject-structure.png" title="Adding a Rest API" width="500" alt="Adding a Rest API"/>

> ### Creating the Integration Logic

Reviewer 2: ```suggestion
### Create the integration logic
```

> #### Creating the First Resource

Reviewer 2: ```suggestion
#### Create the first resource
```

??? note "Source view of the implemented resource"
    ```xml
        <resource methods="POST" uri-template="/create">
            <inSequence>
                <file.write configKey="local_file_connection">
                    <filePath>{json-eval($.filePath)}</filePath>
                    <contentOrExpression>{json-eval($.inputContent)}</contentOrExpression>
                    <mimeType>Automatic</mimeType>
                    <writeMode>Overwrite</writeMode>
                    <appendPosition>0</appendPosition>
                    <encoding>UTF-8</encoding>
                    <compress>false</compress>
                    <appendNewLine>false</appendNewLine>
                    <enableStreaming>false</enableStreaming>
                    <enableLock>false</enableLock>
                    <updateLastModified>true</updateLastModified>
                    <includeResultTo>Message Body</includeResultTo>
                </file.write>
            </inSequence>
            <faultSequence>
            </faultSequence>
        </resource>
    ```

> #### Creating the Second Resource

Reviewer 2: ```suggestion
#### Create the second resource
```

                <default>
                    <log category="INFO" level="simple">
                        <property name="Status" value="File does not exist"/>
                    </log>
                    <drop/>
                </default>
                </switch>
            </inSequence>
            <faultSequence>
            </faultSequence>
        </resource>
    ```

## Get the project

You can download the ZIP file and extract the contents to get the project code.

<a href="{{base_path}}/assets/attachments/connectors/FileConnectorExample.zip">
    <img src="{{base_path}}/assets/img/integrate/connectors/download-zip.png" width="200" alt="Download ZIP">
</a>

## Deployment

Once you have [built your artifacts]({{base_path}}/develop/packaging-artifacts) into a composite application, you can
export it into a CAR file (.car file) using the WSO2 Micro Integrator Visual Studio Code extension:

1.  Open the project overview and click on **Export**.

     <img src="{{base_path}}/assets/img/integrate/connectors/export_artifacts.jpeg" width="300" alt="Download ZIP">

2.  Click on **Select Destination** to select a destination folder to export the CAR file.

> ## Testing

Reviewer 2: ```suggestion
## Test
```

> ### File Create Operation

Reviewer 2: ```suggestion
### File create operation
```

Invoke the API as shown below using the curl command. Curl Application can be downloaded from [here](https://curl.haxx.se/download.html).
```
curl --location 'http://localhost:8290/fileConnector/write' \
--header 'Content-Type: application/json' \
--data '{
"filePath":"/Users/wso2/Documents/mi_testing/doc/create.txt",
"inputContent": "This is a test file"
}'
```

**Expected Response**:

You should get a 'Success' response, and the file should be created in the specified location in the above payload.

> ### File Read Operation

Reviewer 2: ```suggestion
### File read operation
```

Invoke the API as shown below using the curl command. Curl Application can be downloaded from [here](https://curl.haxx.se/download.html).
```
curl --location 'http://localhost:8290/fileConnector/read' \
--header 'Content-Type: application/json' \
--data '{
"filePath":"/Users/wso2/Documents/mi_testing/doc/create.txt"
}'
```

**Expected Response**:

You should get the following text returned.

```
This is a test file.
```

> ## What's Next

Reviewer 2: ```suggestion
## What's next
```
